src/main/java/site/hirecruit/hr/domain/career/service/CareerService.kt

```diff
@@ -0,0 +1,67 @@
+package site.hirecruit.hr.domain.career.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import site.hirecruit.hr.domain.career.CareerEntity
+import site.hirecruit.hr.domain.career.dto.CareerDto
+import site.hirecruit.hr.domain.career.repository.CareerRepository
+import site.hirecruit.hr.domain.company.dto.CompanyDto
+import site.hirecruit.hr.domain.company.repository.CompanyRepository
+import site.hirecruit.hr.domain.user.repository.UserRepository
+import site.hirecruit.hr.global.util.YnType
+
+/**
+ * @author 전지환 이용권&정산유닛 (jyeonjyan@sk.com)
+ */
+@Service
+class CareerService(
+    @Autowired
+    val careerRepository: CareerRepository,
+    @Autowired
+    val companyRepository: CompanyRepository,
+    @Autowired
+    val userRepository: UserRepository
+) {
+
+    fun createCareer(careerCreateRequestDto: CareerDto.CareerCreateRequestDto, userGithubId: Long): CareerDto.Info {
+        val user = userRepository.findByGithubId(userGithubId)
+        val company = companyRepository.findCompanyEntitiesByCompanyId(careerCreateRequestDto.companyId)
+
+        val savedCareer = careerRepository.save(
+            CareerEntity(
+                careerId = null,
+                userEntity = user ?: throw RuntimeException("HR에 회원 정보가 등록되지 않아 처리할 수 없음."),
+                company = company ?: throw RuntimeException("HR에 회사가 등록되지 않아 처리할 수 없음."),
```

Comment:
repository 함수 호출부분에서 null checking하는게 어떤가요?
user가 조회되지 않을 시 company를 조회하지 않아 좀 더 효율적이라 생각이 드네요
```suggestion
        val user = userRepository.findByGithubId(userGithubId)
            ?: throw RuntimeException("HR에 회원 정보가 등록되지 않아 처리할 수 없음.")
        val company = companyRepository.findCompanyEntitiesByCompanyId(careerCreateRequestDto.companyId)
        ?: throw RuntimeException("HR에 회사가 등록되지 않아 처리할 수 없음.")

        val savedCareer = careerRepository.save(
            CareerEntity(
                careerId = null,
                userEntity = user,
                company = company,
```

Reply:
미리 터지게 하자는거죠? 리뷰 감사요